src/ui/public/vislib/lib/handler/handler.js

```diff
@@ -103,7 +103,7 @@ export default function HandlerBaseClass(Private) {
      */
     render() {
       const self = this;
-      const charts = this.charts = [];
+      const { binder, charts = [] } = this;
```

Comment:
fwiw, you can trim the code down by removing these temporary references and just access these from the `self` var. That also has the added advantage that it's clear we're manipulating instance state. you can get of these temporary copies altogether by changing the each-handler to an annonymous function too.